hbase/src/main/java/org/apache/metamodel/hbase/HBaseClient.java

```diff
@@ -0,0 +1,197 @@
+                || values[indexOfIdColumn] == null) {
+            throw new IllegalArgumentException(
+                    "Can't insert a row without having (correct) tableName, columns, values or indexOfIdColumn");
+        }
+        if (columns.length != values.length) {
+            throw new IllegalArgumentException("The amount of columns don't match the amount of values");
+        }
+        try (final Table table = _connection.getTable(TableName.valueOf(tableName))) {
+            // Create a put with the values of indexOfIdColumn as rowkey
+            final Put put = new Put(getValueAsByteArray(values[indexOfIdColumn]));
+
+            // Add the other values to the put
+            for (int i = 0; i < columns.length; i++) {
+                if (i != indexOfIdColumn) {
+                    // NullChecker is already forced within the HBaseColumn class
+                    final byte[] columnFamily = Bytes.toBytes(columns[i].getColumnFamily());
+                    // An HBaseColumn doesn't need a qualifier, this only works when the qualifier is empty (not null).
+                    // Otherwise NullPointer exceptions will happen
+                    byte[] qualifier = null;
+                    if (columns[i].getQualifier() != null) {
+                        qualifier = Bytes.toBytes(columns[i].getQualifier());
+                    } else {
+                        qualifier = Bytes.toBytes(new String(""));
+                    }
+                    final byte[] value = getValueAsByteArray(values[i]);
+                    put.addColumn(columnFamily, qualifier, value);
+                }
+            }
+            // Add the put to the table
+            table.put(put);
+        } catch (IOException e) {
+            throw new MetaModelException(e);
+        }
+    }
+
+    /**
+     * Delete 1 row based on the key
+     * @param tableName
+     * @param rowKey
+     * @throws IllegalArgumentException when any parameter is null
+     * @throws MetaModelException when a {@link IOException} is caught
+     */
+    public void deleteRow(final String tableName, final Object rowKey) {
+        if (tableName == null || rowKey == null) {
+            throw new IllegalArgumentException("Can't delete a row without having tableName or rowKey");
+        }
+        byte[] rowKeyAsByteArray = getValueAsByteArray(rowKey);
+        if (rowKeyAsByteArray.length > 0) {
+            try (final Table table = _connection.getTable(TableName.valueOf(tableName))) {
+                if (rowExists(table, rowKeyAsByteArray)) {
+                    table.delete(new Delete(rowKeyAsByteArray));
+                } else {
+                    logger.warn("Rowkey with value {} doesn't exist in the table", rowKey.toString());
+                }
+            } catch (IOException e) {
+                throw new MetaModelException(e);
+            }
+        } else {
+            logger.info("Have not deleted a row, which has an empty (\"\") rowKey.");
```

Comment:
Should we maybe throw here instead? Or at the least log at error level. But I think throwing makes most sense.